### ISSUE TITLE
Scheduler warning when not needed

### ISSUE THREAD

**Issue by Meehai** [pytorch/pytorch#32347]
## 🐛 Bug

Hello, my high level NN library on top of PyTorch has a bunch of tests. Upon updating to 1.4.0 I wanted to see if my tests still run before diving deeper in potential changes. I have a test that saves and loads a model with scheduler.

More specifically, I get this output:

```
========================================================= test session starts =========================================================
platform linux -- Python 3.8.0, pytest-5.3.2, py-1.8.1, pluggy-0.13.1
rootdir: /home/mihai/Public/Projects/neural-wrappers
collected 13 items                                                                                                                    

test_network.py ............                                                                                                    [ 92%]
test_utils.py .                                                                                                                 [100%]

========================================================== warnings summary ===========================================================
test/pytorch/test_network.py::TestNetwork::test_save_model_2
test/pytorch/test_network.py::TestNetwork::test_save_model_2
test/pytorch/test_network.py::TestNetwork::test_save_model_2
test/pytorch/test_network.py::TestNetwork::test_save_model_2
test/pytorch/test_network.py::TestNetwork::test_save_model_2
test/pytorch/test_network.py::TestNetwork::test_save_model_2
test/pytorch/test_network.py::TestNetwork::test_save_model_2
test/pytorch/test_network.py::TestNetwork::test_save_model_2
test/pytorch/test_network.py::TestNetwork::test_save_model_2
test/pytorch/test_network.py::TestNetwork::test_save_model_2
test/pytorch/test_network.py::TestNetwork::test_save_model_2
test/pytorch/test_network.py::TestNetwork::test_save_model_2
test/pytorch/test_network.py::TestNetwork::test_save_model_2
test/pytorch/test_network.py::TestNetwork::test_save_model_2
test/pytorch/test_network.py::TestNetwork::test_save_model_2
test/pytorch/test_network.py::TestNetwork::test_save_model_2
test/pytorch/test_network.py::TestNetwork::test_save_model_2
test/pytorch/test_network.py::TestNetwork::test_save_model_2
test/pytorch/test_network.py::TestNetwork::test_save_model_2
test/pytorch/test_network.py::TestNetwork::test_save_model_2
test/pytorch/test_network.py::TestNetwork::test_save_model_2
test/pytorch/test_network.py::TestNetwork::test_save_model_2
test/pytorch/test_network.py::TestNetwork::test_save_model_2
test/pytorch/test_network.py::TestNetwork::test_save_model_2
test/pytorch/test_network.py::TestNetwork::test_save_model_2
test/pytorch/test_network.py::TestNetwork::test_save_model_2
test/pytorch/test_network.py::TestNetwork::test_save_model_2
test/pytorch/test_network.py::TestNetwork::test_save_model_2
test/pytorch/test_network.py::TestNetwork::test_save_model_2
test/pytorch/test_network.py::TestNetwork::test_save_model_2
  /usr/lib/python3.8/site-packages/torch/optim/lr_scheduler.py:591: DeprecationWarning: The epoch parameter in `scheduler.step()` was not necessary and is being deprecated where possible. Please use `scheduler.step()` to step the scheduler. During the deprecation, if epoch is different from None, the closed form is used instead of the new chainable form, where available. Please open an issue if you are unable to replicate your use case: https://github.com/pytorch/pytorch/issues/new/choose.
    warnings.warn(EPOCH_DEPRECATION_WARNING, DeprecationWarning)

-- Docs: https://docs.pytest.org/en/latest/warnings.html
=================================================== 13 passed, 30 warnings in 4.67s ===================================================
````

The code that is being run is https://gitlab.com/mihaicristianpirvu/neural-wrappers/blob/master/test/pytorch/test_network.py#L80 and the scheduler is concretely called here https://gitlab.com/mihaicristianpirvu/neural-wrappers/blob/master/neural_wrappers/pytorch/network.py#L246. As you see, I use `step()` and not `step(epoch)` anywhere.

I'm not sure if it's pytest related or anything, I'll do more research and see if I can reproduce.

## Environment

Please copy and paste the output from our
[environment collection script](https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py)
Collecting environment information...
PyTorch version: 1.4.0
Is debug build: No
CUDA used to build PyTorch: 10.1

OS: Manjaro Linux
GCC version: (GCC) 9.2.0
CMake version: version 3.16.0

Python version: 3.8
Is CUDA available: Yes
CUDA runtime version: 10.2.89
GPU models and configuration: GPU 0: GeForce RTX 2070 SUPER
Nvidia driver version: 440.36
cuDNN version: /usr/lib/libcudnn.so.7.6.5

Versions of relevant libraries:
[pip3] numpy==1.17.4
[pip3] torch==1.4.0
[pip3] torchvision==0.2.2.post3
[conda] Could not collect

**Comment by colesbury** [comment]
You call `step(epoch)` here:

https://gitlab.com/mihaicristianpirvu/neural-wrappers/blob/master/neural_wrappers/schedulers.py#L15-16:

```python
	def step(self, epoch=None):
                ...
		epoch = self.model.currentEpoch
		self.baseModel.step(metric, epoch)
```

Please re-open if I'm missing something.
